### PR TITLE
horizontal_space netcdf writer

### PR DESCRIPTION
**Pull request by AlexisRenchon**
The NetCDF writer always tries to interpolate along z, even when the space is a 2D space. This PR ensures that `interpolated_z` is not computed for horizontal spaces.

### CODE REVIEW

File: src/netcdf_writer.jl [hunk 1]
```diff
@@ -142,6 +142,18 @@ function NetCDFWriter(
         )
     end
 
+    if is_horizontal_space
```
Comment (AlexisRenchon) [inline src/netcdf_writer.jl:145]: @Sbozzolo Note we have two conditional statement `if_horizontal_space` in a row, we could possible merge them together? 

We could also move `comms_ctx = ClimaComms.context(space)` above the if statement?

**Review by Sbozzolo — APPROVED** [review]
Please, squash the commits to one before merging.

**Comment by AlexisRenchon** [comment]
@Sbozzolo Do I delete the branch now? (not sure yet what the workflow for merging here is)

**Comment by Sbozzolo** [comment]
Yes, up to you, also please use more informative commit messages next time. Thanks!

**Pull request merged**